src/python/pants/notes/2.1.x.rst

```diff
@@ -0,0 +1,160 @@
+  `PR #11066 <https://github.com/pantsbuild/pants/pull/11066>`_
+
+* Remove deprecated `--protoc-runtime-targets` option (#11083)
+  `PR #11083 <https://github.com/pantsbuild/pants/pull/11083>`_
+
+* Remove deprecated `--interpreter-constraints` option for some Python tools (#11082)
+  `PR #11082 <https://github.com/pantsbuild/pants/pull/11082>`_
+
+* Remove deprecated `runtime_binary_dependencies` field (#11079)
+  `PR #11079 <https://github.com/pantsbuild/pants/pull/11079>`_
+
+* Remove deprecated `--spec-file` option in favor of `--spec-files` (#11076)
+  `PR #11076 <https://github.com/pantsbuild/pants/pull/11076>`_
+
+* Remove deprecated `[basic-auth]`, `[reporter]`, and `[run-tracker]` options (#11078)
+  `PR #11078 <https://github.com/pantsbuild/pants/pull/11078>`_
+
+* Remove deprecated `--print-exception-stacktrace` in favor of `--print-stacktrace` (#11085)
+  `PR #11085 <https://github.com/pantsbuild/pants/pull/11085>`_
+
+* Remove unused global options (#11077)
+  `PR #11077 <https://github.com/pantsbuild/pants/pull/11077>`_
+
+* Remove deprecated `python_requirement` BUILD file symbol (#11055)
+  `PR #11055 <https://github.com/pantsbuild/pants/pull/11055>`_
+
+* Deprecate the `platforms` field allowing a bare string (#11073)
+  `PR #11073 <https://github.com/pantsbuild/pants/pull/11073>`_
+
+* Deprecate several unused options (#11041)
+  `PR #11041 <https://github.com/pantsbuild/pants/pull/11041>`_
+
+* Deprecate unused `run-tracker` options (#11038)
+  `PR #11038 <https://github.com/pantsbuild/pants/pull/11038>`_
+
+* Add `[python-setup].resolver_http_cache_ttl` option (#10997)
+  `PR #10997 <https://github.com/pantsbuild/pants/pull/10997>`_
+
+* Remove deprecated `python-setup` options (#10984)
+  `PR #10984 <https://github.com/pantsbuild/pants/pull/10984>`_
+
+* Remove deprecated `Subsystem.get_options()` in favor of `Subsystem.options` (#10986)
+  `PR #10986 <https://github.com/pantsbuild/pants/pull/10986>`_
+
+* Remove deprecated global options like `--v2` and `--backend-packages2` (#10985)
+  `PR #10985 <https://github.com/pantsbuild/pants/pull/10985>`_
+
+* Remove deprecated `--changed-*` options (#10982)
+  `PR #10982 <https://github.com/pantsbuild/pants/pull/10982>`_
+
+* Remove deprecated `./pants filter` options (#10983)
+  `PR #10983 <https://github.com/pantsbuild/pants/pull/10983>`_
+
+New Features
+~~~~~~~~~~~~
+
+* Add new `py-constraints` goal through new `pants.backend.python.mixed_interpreter_constraints` backend (#11075)
+  `PR #11075 <https://github.com/pantsbuild/pants/pull/11075>`_
+
+* Add `:my_func` shorthand for the `entry_point` field (#11064)
+  `PR #11064 <https://github.com/pantsbuild/pants/pull/11064>`_
+
+* Add new EngineAware method metadata() (#11030)
+  `PR #11030 <https://github.com/pantsbuild/pants/pull/11030>`_
+
+* remote caching support (#10960)
+  `PR #10960 <https://github.com/pantsbuild/pants/pull/10960>`_
+
+* Store per-run logs on RunTracker (#11025)
```

Comment:
Probably internal only?

Reply:
Any pants plugin writer can make use of this feature, even if it's only toolchain using it at the moment.